pandas/core/series.py

```diff
@@ -198,18 +197,41 @@ def __init__(self, data=None, index=None, dtype=None, name=None,
                     data = data.reindex(index, copy=copy)
                 data = data._data
             elif isinstance(data, dict):
-                if index is None:
-                    if isinstance(data, OrderedDict):
-                        index = Index(data)
+                remap_to_mi = False
+                if data:
+                    keys, values = zip(*compat.iteritems(data))
+                    # Workaround for #18485 - part 1/3
```

Comment:
as I said before. this giant block of code is not acceptable here. its needs to be moved to a separate function, and made much simpler.

Reply:
It is smaller than before, but no problem, I'll isolate it. Any comment on the logic?

Reply:
I mean: different than "made much simpler" which is everybody's dream more than a comment.

Reply:
my point is the logic is so dense its not even understandable. this is too complex, no-one will be able to follow this, vet it for bugs or anything. you need to break this apart into logical abstractions, build things up into much smaller units.

Reply:
OK, good!

Reply:
@toobaz don't get me wrong, I am happy to have a fix for this, but it has to be code that is maintainable, limits special cases, and it non completely non-performant.